src/main/kotlin/com/exactpro/th2/ws/client/api/impl/WebSocketClient.kt

```diff
@@ -0,0 +1,224 @@
+/*
+ * Copyright 2021-2021 Exactpro (Exactpro Systems Limited)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.exactpro.th2.ws.client.api.impl
+
+import com.exactpro.th2.common.grpc.Direction
+import com.exactpro.th2.common.grpc.Direction.FIRST
+import com.exactpro.th2.common.grpc.Direction.SECOND
+import com.exactpro.th2.ws.client.api.IClient
+import com.exactpro.th2.ws.client.api.IHandler
+import mu.KotlinLogging
+import java.net.URI
+import java.net.http.HttpClient
+import java.net.http.WebSocket
+import java.nio.ByteBuffer
+import java.util.Base64
+import java.util.concurrent.CompletionStage
+
+class WebSocketClient(
```

Comment:
Please add an event publication in key states.

Reply:
It's already there - we're publishing events for onOpen, onClose, and onError events